lib/suse-cloud-upgrade-4-to-5-post

```diff
@@ -132,6 +132,40 @@ if test -f "$ETC_PROVISIONER_JSON"; then
   fi
 fi
 
+# Populate new directory for repos
+old_repos_dir=/srv/tftpboot/repos
+repos_dir=/srv/tftpboot/suse-11.3/repos
+
+mkdir -p $repos_dir
+
+# Automatically create symlinks for new SMT-mirrored repos if they exist
+for repo in SUSE-Cloud-5-Pool \
+            SUSE-Cloud-5-Updates; do
+  cloud_dir=$repos_dir/$repo
+  smt_dir=/srv/www/htdocs/repo/\$RCE/$repo/sle-11-x86_64
+  test ! -e $cloud_dir -a -d $smt_dir && ln -s $smt_dir $cloud_dir
+done
+
+# For SLE repos, try to re-use the same setup as before, or fallback to
+# symlinks to SMT-mirrored repos
+for repo in SLES11-SP3-Pool \
+            SLES11-SP3-Updates \
+            SLE11-HAE-SP3-Pool \
+            SLE11-HAE-SP3-Updates; do
+  cloud_dir=$repos_dir/$repo
+  smt_dir=/srv/www/htdocs/repo/\$RCE/$repo/sle-11-x86_64
+  if test -L $old_repos_dir/$repo; then
+    ln -s "$(readlink $old_repos_dir/$repo)" $cloud_dir
+  elif test -d $old_repos_dir/$repo; then
+    ln -s $old_repos_dir/$repo $cloud_dir
```

Comment:
Why would we want to copy?  Wouldn't that be hugely slower? 

> I know that some people are short in disk space, even though they really shouldn't

Really shouldn't be short of disk space?  We don't always have any choice, e.g. when running on laptops with small SSDs.